lib/ngx/pipe.lua

```diff
@@ -566,6 +567,38 @@ do
                     error("bad buffer_size option", 2)
                 end
             end
+
+            if opts.environ then
+                environ = opts.environ
+                local environ_type = type(environ)
+                if environ_type ~= 'table' then
+                    error("bad environ option: table expected, got "
+                          .. environ_type, 2)
+                end
+
+                local nenv = 0
+
+                for i, env in ipairs(environ)  do
+                    if env == nil then
+                        break
+                    end
+
+                    nenv = nenv + 1
+
+                    local env_type = type(env)
+                    if env_type ~= 'string' then
+                        error("bad environ option in index " .. i
+                              .. ": string expected, got " .. env_type, 2)
```

Comment:
@thibaultcha 
Environment variable is passed like this format: `KEY=VALUE`. So a `number` argument should not be a valid one.
Maybe we could be more strict, check if there is a `=` in the argument.

Reply:
My bad, while reviewing this particular line I had a hash construct in mind for the `environ` option, which is wrong of course, since the current implementation is an array. Long day.

Such validation is a good idea imho, and adding some test cases with malformed environment strings as well.